File: erlang/pe006.erl
-module(pe006).
-export([start/0]).

maxv() -> 100.

start() ->
  pe006(1, maxv()).

pe006(I, Max) ->
  Sum_sq = sum_sq(I, Max, 0),
  Sq_sum = sq_sum(I, Max),
  io:format("~w~n", [abs(Sq_sum - Sum_sq)]).

ap_sum(Min, Max) ->
  ((Max + Min) * ((Max - Min) + 1)) div 2.

sq_sum(Min, Max) ->
  Sum = ap_sum(Min, Max),
  Sum * Sum.

sum_sq(I, Max, Sum) when I > Max ->
  Sum;

sum_sq(I, Max, Sum) ->
  sum_sq(I + 1, Max, Sum + (I * I)).

File: ocaml/pe006.ml
let maxv = 100 ;;

let ap_sum x y =
  ((x + y) * ((y - x) + 1)) / 2 ;;

let sq_sum x y =
  let z = ap_sum x y in
    z * z ;;

let rec sum_sq i max sum =
  if i > max then
    sum
  else
    sum_sq (i + 1) max (sum + (i * i)) ;;
   
let pe006 i max =
  Printf.printf "%d\n" (abs((sq_sum i max) - (sum_sq i max 0))) ;;

let main () =
  pe006 1 maxv ;;

let () = main ()
